docs/preview/features/openapi/security-definitions.md

```diff
@@ -18,21 +22,59 @@ PM > Install-Package Arcus.WebApi.OpenApi.Extensions
 
 ## Usage
 
-To indicate that an API is protected by OAuth, you need to add `AuthorizeCheckOperationFilter` as an `OperationFilter` when configuring Swashbuckles Swagger generation:
+### OAuth
+
+To indicate that an API is protected by OAuth, you need to add `OAuthAuthorizeOperationFilter` as an `IOperationFilter` when configuring Swashbuckles Swagger generation:
+
+```csharp
+services.AddSwaggerGen(setupAction =>
+{
+    setupAction.SwaggerDoc("v1", new Info { Title = "My API v1", Version = "v1" });
+
+    setupAction.AddSecurityDefinition("oauth2", new OAuth2Scheme
+    {
+        Flow = "implicit",
+        AuthorizationUrl = $"{authorityUrl}connect/authorize",
+        Scopes = scopes
+    });
+    
+    setupAction.OperationFilter<OAuthAuthorizeOperationFilter>(new object[] { new[] { "myApiScope1", "myApiScope2" } });
+});
+```
+
+### Shared Access Key
+
+To indicate that an API is protected by [Shared Access Key authentication](../../features/security/auth/shared-access-key), you need to add `SharedAccessKeyAuthenticationOperationFilter` as an `IOperationFilter` when configuring Swashbuckles Swagger generation:
+
+```csharp
+services.AddSwaggerGen(setupAction =>
+{
+    setupAction.SwaggerDoc("v1", new Info { Title = "My API v1", Version = "v1" });
+
+    setupAction.AddSecurityDefinition("sharedaccesskey", new OpenApiSecurityScheme
+    {
+        Type = SecuritySchemeType.ApiKey
+     });
+
+     setupAction.OperationFilter<SharedAccessKeyAuthenticationOperationFilter>(new object[] { new[] { "myApiScope1", "myApiScope2" } });
+});
+```
+
+### Shared Access Key
+
+To indicate that an API is protected by [Certificate authentication](../../features/security/auth/certificate), you need to add `CertificateAuthenticationOperationFilter` as an `IOperationFilter` when configuring Swashbuckles Swagger generation:
 
 ```csharp
 services.AddSwaggerGen(setupAction =>
 {
-   setupAction.SwaggerDoc("v1", new Info { Title = "My API v1", Version = "v1" });
+    setupAction.SwaggerDoc("v1", new Info { Title = "My API v1", Version = "v1" });
 
-   setupAction.AddSecurityDefinition("oauth2", new OAuth2Scheme
-   {
-      Flow = "implicit",
-      AuthorizationUrl = $"{authorityUrl}connect/authorize",
-      Scopes = scopes
-   });
+    setupAction.AddSecurityDefinition("certificate", new OpenApiSecurityScheme
+    {
+        Type = SecuritySchemeType.ApiKey
```

Comment:
Copy paste?

Reply:
Looks like this is coming as part of 3.1 - https://github.com/OAI/OpenAPI-Specification/releases/tag/3.1.0-rc0

Not sure what to use then, any thoughts @fgheysels @gverstraete ?

See https://github.com/OAI/OpenAPI-Specification/issues/1004#issuecomment-332894619

Reply:
@fgheysels is on leave, @gverstraete can maybe help us.

Reply:
3.0 is EOL right? so I would do it the optimal way only for 3.1

Reply:
This is OpenAPI 3.0 which is the latest version; not .NET Core if you are talking about that @gverstraete ?

Reply:
Ok; was too fast in replying :) Sorry! Then I think ApiKey is the only option here. What would you suggest otherwise..

Reply:
Good point 😂